src/Bicep.Core/Analyzers/Linter/LinterRuleBase.cs

```diff
@@ -77,7 +77,25 @@ protected virtual string GetFormattedMessage(params object[] values)
             return this.Description;
         }
 
-        public abstract IEnumerable<IBicepAnalyzerDiagnostic> Analyze(SemanticModel model);
+        public IEnumerable<IBicepAnalyzerDiagnostic> Analyze(SemanticModel model)
+        {
+            try
+            {
+                return AnalyzeInternal(model);
+            }
+            catch (Exception ex)
+            {
+                return new[]{ new AnalyzerDiagnostic(this.AnalyzerName,
```

Comment:
This catch block doesn't work If you put an exception at the end of a rule's AnalyzeInternal. The try/catch is only valid for the return of the iterator.  Exceptions in the rule actually occur during the iteration, and are caught as a Fatal compilation error:

![image](https://user-images.githubusercontent.com/6913354/118028384-dd19b000-b317-11eb-8b4f-9e65e29dee9c.png)
here:
```cs
        private (ImmutableArray<SyntaxTree> added, ImmutableArray<SyntaxTree> removed) UpdateCompilationInternal(DocumentUri documentUri, int? version)
        {
            try
            {
 ...
            catch (Exception exception)
            {
                var fatalError = new OmniSharp.Extensions.LanguageServer.Protocol.Models.Diagnostic
             ...
```

Reply:
And good, your unit tests catch this.  Easiest fix is to ensure the enumerator is run during the try/catch:

return AnalyzeInternal(model).ToArray();

Reply:
You unit tests didn't catch this because your fake rule wasn't returning an iterator.  I've got a fix.

Reply:
https://github.com/jfleisher/bicep/pull/2

Reply:
Actually, this fix isn't sufficient, it won't handle external linters.  Added a TODO for now.

Reply:
If you agree this is good enough for now, we can resolve.